mesh/mesh.go

```diff
@@ -389,6 +349,46 @@ func (vl *validator) ValidateLayer(ctx context.Context, lyr *types.Layer) {
 	logger.Info("done validating layer")
 }
 
+func (msh *Mesh) persistLayerHashes(ctx context.Context, from, to types.LayerID) error {
+	if to.Before(from) {
+		msh.WithContext(ctx).With().Panic("verified layer went backward",
+			log.FieldNamed("fromLayer", from),
+			log.FieldNamed("toLayer", to))
+	}
+
+	msh.WithContext(ctx).With().Debug("persisting layer hashes",
+		log.FieldNamed("from_layer", from),
+		log.FieldNamed("to_layer", to))
+	for i := from; !i.After(to); i = i.Add(1) {
+		lyr, err := msh.GetLayer(i)
+		if err != nil {
+			msh.WithContext(ctx).With().Error("failed to get verified layer", i, log.Err(err))
+			return err
+		}
+		validBlocks, _ := msh.BlocksByValidity(lyr.Blocks())
+		sorted := types.SortBlockIDs(types.BlockIDs(validBlocks))
```

Comment:
blocks are stored in the sorted order in the database and returned that way during iteration

Reply:
i hesitate. the underlying db impl may change over time. i'd prefer to keep this sort to guarantee its deterministic outcome. 
in the future we will have unified content block so i don't think performance will be the issue.

Reply:
i suppose that it can be ensured with unit tests. db usually good at storing things in sorted order, so i won't worry much that it will change, but up to you.

Reply:
adding a test is a good idea. done.